metadata/metadata.go

```diff
@@ -1444,7 +1445,17 @@ func (serv *MetadataServer) ListProviders(_ *pb.Empty, stream pb.Metadata_ListPr
 }
 
 func (serv *MetadataServer) CreateProvider(ctx context.Context, provider *pb.Provider) (*pb.Empty, error) {
-	return serv.genericCreate(ctx, &providerResource{provider}, nil)
+	// hack until we have a nice abstraction for the config updates
+	provResource := &providerResource{provider}
```

Comment:
this doesn't ultimately belong here but it's a hack until we get things figured out correctly